jpf-symbc/src/main/gov/nasa/jpf/symbc/bytecode/SymbolicStringHandler.java

```diff
@@ -1172,6 +1354,106 @@ public Instruction handleSubString2(JVMInvokeInstruction invInst, ThreadInfo th)
 			sf.push(objRef.getObjectRef(), true);
 			//System.out.println("[SymbolicStringHandler] " + sf.toString());
 			sf.setOperandAttr(result);
+            } else if (currentChocie == 4) {
+                if (!rte_flag) {
+                    th.getVM().getSystemState().setIgnored(true);
+                } else {
+                    if(sym_v2 != null) {
+                        if(sym_v1 != null) {
+                            pc._addDet(Comparator.GT, sym_v2, sym_v1);
+                        } else {
+                            int val = s1;
+                            pc._addDet(Comparator.GT, sym_v2, new IntegerConstant(val));
+                        }
+                    } else {
+                        int val2 = s2;
+                        pc._addDet(Comparator.GT, new IntegerConstant(val2), sym_v1);
+                    }
+                    if (!pc.simplify()) {
+                        th.getVM().getSystemState().setIgnored(true);
+                    } else {
+                        th.createAndThrowException("java.lang.StringIndexOutOfBoundsException");
```

Comment:
you need to think about some code restructuring because that is a lot of code to have in one method.
Also, I am not sure why you need 6 choices (from 0-5)? This must be wrong.